pyswip/core.py

```diff
@@ -583,16 +583,30 @@ def check_and_call(*args):
 PL_VERSION_VM		=6	# VM signature
 PL_VERSION_BUILT_IN	=7	# Built-in predicate signature
 
+
+# After SWI-Prolog 8.5.2, PL_version was renamed to PL_version_info
+# to avoid a conflict with Perl. For more details, see the following:
+# https://github.com/SWI-Prolog/swipl-devel/issues/900
+# https://github.com/SWI-Prolog/swipl-devel/issues/910
 try:
-    PL_version = _lib.PL_version
-    PL_version.argtypes = [c_int]
-    PL_version.restype = c_uint
+    # swi-prolog > 8.5.2:
+    PL_version_info = _lib.PL_version_info
+    PL_version_info.argtypes = [c_int]
+    PL_version_info.restype = c_uint
 
-    PL_VERSION = PL_version(PL_VERSION_SYSTEM)
-    if PL_VERSION<80200:
-        raise Exception("swi-prolog>= 8.2.0 is required")
+    PL_VERSION = PL_version_info(PL_VERSION_SYSTEM)
 except AttributeError:
-    PL_VERSION=70000  # Best guess. When was PL_version introduced?
+    # swi-prolog <= 8.5.2:
+    try:
+        PL_version = _lib.PL_version
```

Comment:
I use Aleph and it wasn't working with SWI_prolog 9.x until I changed this line to `PL_version = _lib.PL_version_info`.

Reply:
This works for me, thanks@AmrSol